Sources/SPTDataLoader/SPTDataLoaderService.m

```diff
@@ -41,9 +42,11 @@ @interface SPTDataLoaderService () <SPTDataLoaderRequestResponseHandlerDelegate,
 @property (nonatomic, strong) NSOperationQueue *sessionQueue;
 @property (nonatomic, strong) NSMutableArray<SPTDataLoaderRequestTaskHandler *> *handlers;
 @property (nonatomic, strong) NSMapTable<id<SPTDataLoaderConsumptionObserver>, dispatch_queue_t> *consumptionObservers;
+@property (nonatomic, strong, nullable, readonly) id<SPTDataLoaderCrashReporter> crashReporter;
 @property (nonatomic, strong) SPTDataLoaderServerTrustPolicy *serverTrustPolicy;
 @property (nonatomic, weak, nullable) NSFileManager *fileManager;
 @property (nonatomic, weak, nullable) Class dataClass;
+@property (nonatomic, strong) NSMutableSet<NSURLSession *> *invalidatedSessions;
```

Comment:
I would avoid to add this complexity in keeping track of what session got invalidated.
The `invalidateAndCancel` method always invalidates all the sessions handled by the SessionSelector.
My suggestion is to keep only a boolean to keep track of the session invalidation. This will simplify this PR